actionpack/test/dispatch/content_security_policy_test.rb

```diff
@@ -302,6 +302,73 @@ def test_raises_runtime_error_when_unexpected_source
   end
 end
 
+class ContentSecurityPolicyMiddlewareTest < ActiveSupport::TestCase
+  def test_rack_lint
+    app = proc { [200, {}, []] }
+    env = Rack::MockRequest.env_for("", {})
+
+    assert_nothing_raised do
+      Rack::Lint.new(
+        ActionDispatch::ContentSecurityPolicy::Middleware.new(
+          Rack::Lint.new(app)
+        )
+      ).call(env)
+    end
+  end
+
+  def test_does_not_override_app_content_security_policy
+    app = proc { [200, { "content-security-policy" => "default-src 'self' https: http:" }, []] }
+    env = Rack::MockRequest.env_for("", {})
+
+    env["action_dispatch.content_security_policy"] = ActionDispatch::ContentSecurityPolicy.new do |p|
+      p.default_src -> { :self  }
+    end
+    env["action_dispatch.content_security_policy_nonce_generator"] = proc { "iyhD0Yc0W+c=" }
+    env["action_dispatch.content_security_policy_report_only"] = false
+    env["action_dispatch.show_exceptions"] = :none
```

Comment:
Maybe we could pull all this env setup into a `setup` helper for this test class?

Reply:
Done!